sdk/python/lib/pulumi/provider/experimental/provider.py

```diff
@@ -25,6 +25,9 @@
 from .schema import generate_schema
 
 
+class MissingPropertyError(Exception): ...
```

Comment:
I think I can use `InputPropertiesError` here for a nicer error message via https://github.com/pulumi/pulumi/blob/96a351b687c687e10ca6617419205e6c220e7b43/sdk/python/lib/pulumi/provider/server.py#L115C16-L117

Reply:
Yeah, I think `InputPropertiesError` would get translated nicely to diagnostics and shown to the user in a nice way.